src/types.ts

```diff
@@ -3,3 +3,5 @@ interface ProviderClass<T = any> extends Function {
 }
 
 export type { ProviderClass };
+
+export type IScope = 'default' | 'non-singleton';
```

Comment:
Same as above, let's move this up and use the existing `export type` to export it.